neptune_tensorflow_keras/impl/__init__.py

```diff
@@ -120,20 +119,29 @@ def _log_metrics(self, logs, category: str, trigger: str):
 
         for metric, value in logs.items():
             try:
-                if metric in ('batch', 'size') or metric.startswith('val_'):
+                if metric in ("batch", "size") or metric.startswith("val_"):
                     continue
                 logger[metric].log(value)
             except NeptuneException:
                 pass
 
+    def on_train_begin(self, logs=None):  # pylint:disable=unused-argument
+        self._metric_logger["model/summary"] = _model_summary_file(self.model)
+
     def on_train_batch_end(self, batch, logs=None):  # pylint:disable=unused-argument
-        self._log_metrics(logs, 'train', 'batch')
+        self._log_metrics(logs, "train", "batch")
 
     def on_epoch_end(self, epoch, logs=None):  # pylint:disable=unused-argument
-        self._log_metrics(logs, 'train', 'epoch')
+        self._log_metrics(logs, "train", "epoch")
 
     def on_test_batch_end(self, batch, logs=None):  # pylint:disable=unused-argument
-        self._log_metrics(logs, 'test', 'batch')
+        self._log_metrics(logs, "test", "batch")
+
+    def on_test_end(self, logs=None):  # pylint:disable=unused-argument
+        self._log_metrics(logs, "test", "epoch")
+
 
-    def on_test_end(self, logs=None):
-        self._log_metrics(logs, 'test', 'epoch')
+def _model_summary_file(model) -> File:
+    stream = io.StringIO()
+    model.summary(print_fn=lambda x: stream.write(x + "\n"))
+    return File.from_stream(stream, extension="txt")
```

Comment:
Are you sure txt is good extension for model?

Reply:
It's the result of Python's `print`, it wouldn't need an extension, but with txt we get a preview.

Reply:
Isn't it a json/xml or some other well known format that we can get preview of?

Reply:
@aniezurawski nope, it's something like this:

```
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 flatten (Flatten)           (None, 784)               0         
                                                                 
 dense (Dense)               (None, 10)                7850      
                                                                 
=================================================================
Total params: 7,850
Trainable params: 7,850
Non-trainable params: 0
_________________________________________________________________
```

Reply:
Hah. I imagine trying to parse such thing is nightmare. :P
Ok, fine. :)